Neos.Utility.ObjectHandling/Classes/TypeHandling.php

```diff
@@ -111,7 +111,18 @@ public static function isLiteral(string $type): bool
      */
     public static function isSimpleType(string $type): bool
     {
-        return in_array(self::normalizeType($type), ['array', 'string', 'float', 'integer', 'boolean'], true);
+        return in_array(self::normalizeType($type), ['array', 'string', 'float', 'integer', 'boolean', 'null'], true);
+    }
+
+    /**
+     * Returns true if the $type is a literal type.
+     *
+     * @param string $type
+     * @return boolean
+     */
+    public static function isLiteralType(string $type): bool
+    {
+        return in_array($type, ['false', 'true'], true);
```

Comment:
I'm confused by the semantics here. The reason is basically that I don't know what the idea of "simple type" (which has not been introduced in this PR) refers to.

There are three methods `isLiteral`, `isSimpleType` and `isLiteralType`. Looking at:
https://github.com/neos/flow-development-collection/blob/7d02951763753b9ff80bb6b4a478b20631e76059/Neos.Utility.ObjectHandling/Classes/TypeHandling.php#L29-L32

I assume that `isLiteral` and `isSimpleType` address the same set of types (in that case, the `LITERAL_TYPE_PATTERN` would need to be adjusted to cover `null` as well).

But I'd like to suggest another way:

Maybe this would be a good opportunity to adjust the `TypeHandling` class to operate closer to PHP's type system definitions (as per https://www.php.net/manual/en/language.types.type-system.php). I'd suggest the following methods:

* `isScalarType`
  * covers `bool`, `int`, `float` and `string`
* `isBuiltInType`
  * covers `null`, `array`, `object`, `resource`, `never`, `void`, `self`, `parent` and `static`
  * falls back to `isScalarType`
* `isLiteralType`
  * Same as it is right now in this PR
* `isCallableType`
  * covers `callable`
* `isBuiltInTypeAlias`
  * covers `mixed` and `iterable`
* `isUserDefinedType`
  * calculated from: `!isBuiltInType && !isLiteralType && !isCallableType && !isBuiltInTypeAlias`
  * can be used in `ReflectionService` to determine whether the type name needs to be prefixed with `\` or not

For backwards compatibility `isLiteral` and `isSimpleType` need to continue to work, but can use the new methods. Given my suggestion, `isLiteral` and `isSimpleType` should (probably) also be deprecated.

Reply:
@grebaldi the idea of literal type comes from php itself https://www.php.net/manual/de/language.types.type-system.php but I find it weird myself and would be fine with just adding true false null to the simple types.